tests/test_processing_common.py

```diff
@@ -86,9 +87,22 @@ def get_processor(self):
         processor = self.processor_class(**components, **self.prepare_processor_dict())
         return processor
 
+    def prepare_text_inputs(self, batch_size: Optional[int] = None):
+        if batch_size is not None:
+            if batch_size < 1:
+                raise ValueError("batch_size must be greater than 0")
+            elif batch_size == 1:
+                return ["lower newer"]
+            return ["lower newer", "upper older longer string"] + ["lower newer"] * (batch_size - 2)
+        return "lower newer"
```

Comment:
nit - it's better to do fast returns and keep most of the logic unindented. The input validation is also maybe a bit overkill as this is just for tests but never bad to have :) 

```suggestion
        if batch_size is None:
            return "lower newer"
        
        if batch_size < 1:
            raise ValueError("batch_size must be greater than 0")
        
        if batch_size == 1:
            return ["lower newer"]
        return ["lower newer", "upper older longer string"] + ["lower newer"] * (batch_size - 2)
```